### PR TITLE
Bump omniauth gem to ~> 2.0

### PR DESCRIPTION
**Pull request by sineed**
- add omniauth csrf protection to make sure auth requests are going via POST request
- add repost gem to do post requests and keep existing flow (user doesn't ckick anything)

### CODE REVIEW

File: bookingsync-engine.gemspec [hunk 1]
```diff
@@ -17,6 +17,8 @@ Gem::Specification.new do |s|
 
   s.add_dependency "rails", ">= 5.0.0"
   s.add_dependency "omniauth-bookingsync", '>= 0.5.0'
+  s.add_dependency "omniauth-rails_csrf_protection", "~> 1.0"
+  s.add_dependency "repost"
```
Comment (LeoMo42) [inline bookingsync-engine.gemspec:21]: do you think it worth to specify the version here?

**Review by LeoMo42 — APPROVED** [review]

File: bookingsync-engine.gemspec (hunk 1 above)
Comment (sineed) [reply bookingsync-engine.gemspec:21]: makes sense

**Comment by sineed** [comment]
@LeoMo42 I'm not sure in terms of SemVer what to set as the next version after this will be pushed to master. It will totally break all consumers which are not ready for this fix. So it has to be `6.0.0`. But I'm not sure. Perhaps I may leave it in master branch and use it without version then.
Any suggestions here?

**Comment by LeoMo42** [comment]
not sure this is really so significant change to bump major version.

I think we should bump minor version only.

@StoneFrog , @Azdaroth WDYT guys?

**Comment by sineed** [comment]
It is not significant, but it is breaking. But I think that we may wait more commits and then bump the version. From this time it can be utilized from the `master` branch if you want to bump omniauth in your project

**Comment by Azdaroth** [comment]
If it's a breaking change, I would surely  bump by a major version

**Pull request merged**